src/Adapter/MSTestAdapter.PlatformServices/ObjectModel/UnitTestElement.cs

```diff
@@ -277,7 +277,8 @@ private Guid GenerateSerializedDataStrategyTestId(string testFullName)
 
 #if NET9_0_OR_GREATER
         // Version property is only available on .NET 9 and later.
-        Debug.Assert(guid.Version == 8, "Expected Guid version to be 8");
+        Debug.Assert(guid.Version == 8, $"Expected Guid version to be 8, but it was {guid.Version}");
+        Debug.Assert(guid.Variant == 8, $"Expected Guid variant to be 8, but it was {guid.Variant}");
```

Comment:
Isn't variant supposed to be 2 (`0b10`)?

Reply:
Spec:

> The 2-bit variant field as defined by [Section 4.1](https://www.rfc-editor.org/rfc/rfc9562.html#variant_field), set to 0b10. Occupies bits 64 and 65 of octet 8.

.NET implementation is so weird then. 👀

Reply:
https://github.com/dotnet/runtime/pull/118987